tests/unit/modules/test_freebsd_sysctl.py

```diff
@@ -0,0 +1,147 @@
+        with patch.dict(freebsd_sysctl.__salt__, {"cmd.run_all": mock_cmd}):
+            self.assertRaises(
+                CommandExecutionError,
+                freebsd_sysctl.assign,
+                "asef.esrhaseras.easr",
+                "backward",
+            )
+
+    def test_assign_success(self):
+        """
+        Tests the return of successful assign function
+        """
+        cmd = {
+            "pid": 1337,
+            "retcode": 0,
+            "stderr": "",
+            "stdout": "vfs.usermount: 0 -> 1",
+        }
+        ret = {"vfs.usermount": "1"}
+        mock_cmd = MagicMock(return_value=cmd)
+        with patch.dict(freebsd_sysctl.__salt__, {"cmd.run_all": mock_cmd}):
+            self.assertEqual(freebsd_sysctl.assign("vfs.usermount", 1), ret)
+
+    def test_persist_no_conf_failure(self):
+        """
+        Tests adding of config file failure
+        """
+        asn_cmd = {
+            "pid": 1337,
+            "retcode": 1,
+            "stderr": "sysctl: vfs.usermount=1: Operation not permitted",
+            "stdout": "vfs.usermount: 1",
+        }
+        mock_asn_cmd = MagicMock(return_value=asn_cmd)
+        cmd = "sysctl vfs.usermount=1"
+        mock_cmd = MagicMock(return_value=cmd)
+        with patch.dict(
+            freebsd_sysctl.__salt__,
+            {"cmd.run_stdout": mock_cmd, "cmd.run_all": mock_asn_cmd},
+        ):
+            with patch("salt.utils.files.fopen", mock_open()) as m_open:
+                self.assertRaises(
+                    CommandExecutionError,
+                    freebsd_sysctl.persist,
+                    "net.ipv4.ip_forward",
+                    1,
+                    config=None,
+                )
+
+    def test_persist_updated(self):
+        """
+        Tests sysctl.conf success
+        """
+        cmd = {
+            "pid": 1337,
+            "retcode": 0,
+            "stderr": "",
+            "stdout": "vfs.usermount: 1 -> 1",
+        }
+        mock_cmd = MagicMock(return_value=cmd)
+
+        with patch("salt.utils.files.fopen", mock_open()):
+            with patch.dict(
+                freebsd_sysctl.__salt__, {"cmd.run_all": mock_cmd},
+            ):
+                self.assertEqual(
+                    freebsd_sysctl.persist("vfs.usermount", 1), "Updated",
+                )
+
+    def test_persist_updated_tunable(self):
+        """
+        Tests loader.conf success
+        """
+
+        with patch("salt.utils.files.fopen", mock_open()):
+            self.assertEqual(
+                freebsd_sysctl.persist("vfs.usermount", 1, "/boot/loader.conf"),
+                "Updated",
+            )
+
+    def test_show(self):
+        """
+        Tests the show function
+        """
+        # Mock just a small portion of the full "sysctl -ae" output, but be
+        # sure to include a multi-line value.
+        mock_cmd = MagicMock(
+            return_value="""kern.ostype=FreeBSD
```

Comment:
Please use `tests.supper.helpers.wrap` for `return_value`

Reply:
Was that `tests.support.helpers.wrap`?  Though I'm not seeing that :thinking: - or are you referring to textwrap?

Reply:
Sorry `tests.support.helpers.dedent`